index.js

```diff
@@ -2,20 +2,29 @@
 
 var electron = require('electron');
 var BrowserWindow = electron.BrowserWindow || electron.remote.BrowserWindow;
+var nativeImage = electron.nativeImage;
 
 module.exports = function electronImageResize(params) {
   var opts = params || {};
   return new Promise((resolve, reject) => {
+
     if (typeof opts.url !== 'string') {
       reject(new TypeError('Expected option: `url` of type string'));
     }
 
+    if (typeof opts.height !== 'number' && typeof opts.width !== 'number') {
+      reject(new TypeError('Expected option: `height` or `width` of type number'));
+    }
+
+    var imageLocation = opts.url.replace('file://', '');
+    var originalSize = nativeImage.createFromPath(imageLocation).getSize();
```

Comment:
Can you add a conditional to this so that we're not getting the size of the image if we don't need to. If both `width` and `height` are specified then we shouldn't need to get the original size.